lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -473,7 +564,7 @@ fn split_at_delimiter(input: &[u8], delimiter: u8) -> Option<(&[u8], &[u8])> {
 
 #[inline]
 fn ascii_alphanum_and_seps(input: &[u8]) -> IResult<&[u8], &str> {
-    let valid_char = |c: u8| c.is_ascii_alphanumeric() || c == b'_' || c == b'-' || c == b'.';
+    let valid_char = |c: u8| c.is_ascii_alphanumeric() || c == b' ' || c == b'_' || c == b'-' || c == b'.';
```

Comment:
I added `b' '` so that service check messages can contain spaces. Also, this function used to be called `metric_name` since it was only used for parsing metric names. Metric names can have spaces (they just get converted to spaces) `statsd.increment('raymond space') => raymond_space` in dd

Reply:
Do spaces in metric names get converted in the Agent itself on ingest/decode? If so, seems like we might need to open an issue to handle that... 😅

Reply:
Conversion happens in the intake side.

Reply:
Phew. 😅